agasc/supplement/magnitudes/mag_estimate.py

```diff
@@ -0,0 +1,905 @@
+        'n_ok': np.sum(ok),
+        'outliers': -1,
+        'lf_variability_100s': np.inf,
+        'lf_variability_500s': np.inf,
+        'lf_variability_1000s': np.inf,
+        'tempccd': np.nan,
+        'dr_star': dr_star,
+    }
+    if stats['n_ok'] < 10:
+        return stats
+
+    aoacmag_q25, aoacmag_q50, aoacmag_q75 = np.quantile(telem['AOACMAG'][ok], [0.25, 0.5, 0.75])
+
+    mags = telem['mags']
+    q25, q50, q75 = np.quantile(mags[ok], [0.25, 0.5, 0.75])
+    iqr = q75 - q25
+    outlier = ok & ((mags > q75 + 3 * iqr) | (mags < q25 - 3 * iqr))
+
+    s_100s = rolling_mean(times, mags, window=100, selection=ok & ~outlier)
+    s_500s = rolling_mean(times, mags, window=500, selection=ok & ~outlier)
+    s_1000s = rolling_mean(times, mags, window=1000, selection=ok & ~outlier)
+
+    s_100s = s_100s[~np.isnan(s_100s)]
+    s_500s = s_500s[~np.isnan(s_500s)]
+    s_1000s = s_1000s[~np.isnan(s_1000s)]
+
+    stats.update({
+        'aoacmag_mean': np.mean(telem['AOACMAG'][ok]),
+        'aoacmag_err': np.std(telem['AOACMAG'][ok]),
+        'aoacmag_q25': aoacmag_q25,
+        'aoacmag_median': aoacmag_q50,
+        'aoacmag_q75': aoacmag_q75,
+        'q25': q25,
+        'median': q50,
+        'q75': q75,
+        'counts_img': np.mean(telem['counts_img'][ok]),
+        'counts_dark': np.mean(telem['counts_dark'][ok]),
+        'mean': np.mean(mags[ok]),
+        'mean_err': scipy.stats.sem(mags[ok]),
+        'std': np.std(mags[ok]),
+        'skew': scipy.stats.skew(mags),
+        'kurt': scipy.stats.kurtosis(mags),
+        't_mean': np.mean(mags[ok & (~outlier)]),
+        't_mean_err': scipy.stats.sem(mags[ok & (~outlier)]),
+        't_std': np.std(mags[ok & (~outlier)]),
+        't_skew': scipy.stats.skew(mags[ok & (~outlier)]),
+        't_kurt': scipy.stats.kurtosis(mags[ok & (~outlier)]),
+        'outliers': np.sum(outlier),
+        'lf_variability_100s': np.max(s_100s) - np.min(s_100s),
+        'lf_variability_500s': np.max(s_500s) - np.min(s_500s),
+        'lf_variability_1000s': np.max(s_1000s) - np.min(s_1000s),
+        'tempccd': np.mean(telem['TEMPCCD'][ok]) - 273.16,
+    })
+
+    stats.update({
+        'mag_img': np.mean(telem['mags_img'][ok & (~outlier)]),
+        'mag_obs': stats['t_mean'],
+        'mag_obs_err': stats['t_mean_err']
+    })
+
+    return stats
+
+
+def get_agasc_id_stats(agasc_id, obs_status_override=None, tstop=None):
+    """
+    Get summary magnitude statistics for an AGASC ID.
+
+    :param agasc_id: int
+    :param obs_status_override: dict.
+        Dictionary overriding the OK flag for specific observations.
+        Keys are (OBSID, AGASC ID) pairs, values are dictionaries like
+        {'obs_ok': True, 'comments': 'some comment'}
+    :param tstop: cxotime-compatible timestamp
+        Only entries in catalogs.STARS_OBS prior to this timestamp are considered.
+    :return: dict
+        dictionary with stats
+    """
+    min_mag_obs_err = 0.3
```

Comment:
Maybe this is the bit that needs to be 0.03?

Reply:
Fixed.